ols/user_data_collection/data_collector.py

```diff
@@ -0,0 +1,303 @@
+"""Collect insights and upload it to the Ingress service.
+
+It waits 5 min after startup before collects the data. Then it collects
+data after specified interval.
+
+When CP_OFFLINE_TOKEN is provided (either for prod or stage), it is used
+for ingress authentication instead of cluster pull-secret.
+"""
+
+import base64
+import io
+import json
+import logging
+import os
+import pathlib
+import sys
+import tarfile
+import time
+from typing import Any, Callable
+
+import kubernetes
+import requests
+
+OLS_USER_DATA_PATH = os.environ["OLS_USER_DATA_PATH"]
```

Comment:
When we use config, we also need to mount volumes with credentials, otherwise it will fail as config expects it.

Updating a sys path is not related only to config only, but it is a more general "problem" because the ols itself is not installed (just run via uvicorn). So in every script where something from ols is needed, the sys path needs to be updated with the project root.

The collector needs to know the root of the data dir. Currently, we set the specific paths in the config for feedback/transcripts dirs. Then in collector code we would need to do a bit of checking/ensuring the paths from where data should be collected.
It is a bit inconsistent how data paths are defined in config and what script expects (all data are stored in a single root). We should either set a path for user data in config (instead of specifying for feedback/transcripts) or explicitly use both paths in collector.

Follow-up https://issues.redhat.com/browse/OLS-514.

Reply:
sounds good, thanks.